### PR TITLE
fix(slider): fix `Home/End/PageDown/PageUp` keys

### PR DESCRIPTION
**Pull request by royeden**
Fixed the following issues with the [`Slider`](https://kobalte.dev/docs/core/components/slider):

* When using  `Home/End/PageDown/PageUp` keys, the page would scroll instead of staying in place and updating the slider values. This is not the intended behavior, so the default event is prevented and the propagation is stopped to fix it. This was reused in many places across the slider, so I abstracted it into a `util` called `stopEventDefaultAndPropagation` (feel free to ask me to bring back the old behavior with less function calls).

* Slider behavior wasn't aligned with the WAI-ARIA Multi-Thumb slider: When you have a [Multi-Thumb Slider](https://www.w3.org/WAI/ARIA/apg/patterns/slider-multithumb/examples/slider-multithumb/) and you press the `Home/End` key, the expectation is for the currently focused thumb to go as far as it's allowed, but the current implementation simply forced the `Home` key to snap the first thumb at the min value and the `End` key to snap the second thumb to the max value. This PR addresses that by grabbing the `min/max` values for the focused thumb instead of a fixed index. I also made the `getThumbMinValue/getThumbMaxValue` functions aware of the `minStepsBetweenThumbs`.

* Simplifies increment / decrement thumb: abstracted their repeated code into a function called `snapThumbValue` (same mention here regarding bringing back the old behavior for less function calls).

* Removed unused import of `getClosestValueIndex` in `slider-root.tsx`.

* Simplified the fall-through cases for the `switch` statement that handles the slider keys.

* Fixed `getClosestValueIndex` to resolve ambigous values.

* Focus the correct thumb after `onSlideEnd`.

~A small comment that I've noticed and currently cannot wrap my head around: Why are there 2 focus points between sliders (both the thumb and the hidden input are focusable, creating the need for two `tab` key presses before switching focus)? This issue is not present in [`React-ARIA`'s `RangeSlider`](https://react-spectrum.adobe.com/react-spectrum/RangeSlider.html#rangeslider). I think this could also be addressed setting `tabIndex={-1}` for the inputs.
Let me know if this PR should also incorporate that change.~

Thanks as always for such a nice library!

### CODE REVIEW

**Comment by fabien-ml** [comment]
Hi, thanks for the PR.
@dev-rb I'll wait for your approbation before merge.

File: packages/core/src/slider/create-slider-state.ts [hunk 1]
```diff
@@ -121,11 +121,17 @@ export function createSliderState(props: StateOpts): SliderState {
...
+      : values()[index - 1] +
+          props.minStepsBetweenThumbs!() * props.step!() * (values().length - 1);
   };
 
   const getThumbMaxValue = (index: number) => {
-    return index === values().length - 1 ? props.maxValue!() : values()[index + 1];
+    return index === values().length - 1
+      ? props.maxValue!()
+      : values()[index + 1] -
+          props.minStepsBetweenThumbs!() * props.step!() * (values().length - 1);
```
Comment (dev-rb) [inline packages/core/src/slider/create-slider-state.ts:134]: Adding the * (values().length - 1) seems to not create the correct result. I tested this with 3 thumbs, selecting the middle thumb, and using the Home and End keys to set the thumb to it's min/max values.
```suggestion
          props.minStepsBetweenThumbs!() * props.step!()
```

Comment (dev-rb) [reply packages/core/src/slider/create-slider-state.ts:134]: I think if you can change the `getThumbMinValue` and `getThumbMaxValue` as suggested above or let me know if I am misunderstanding something, it looks good to me. 

Regarding your last comment, I hadn't noticed that issue. Feel free to include it in the PR as well.

Thanks so much for the PR @royeden!

File: packages/core/src/slider/create-slider-state.ts [hunk 2]
```diff
@@ -121,11 +121,17 @@ export function createSliderState(props: StateOpts): SliderState {
   };
 
   const getThumbMinValue = (index: number) => {
-    return index === 0 ? props.minValue!() : values()[index - 1];
+    return index === 0
+      ? props.minValue!()
+      : values()[index - 1] +
+          props.minStepsBetweenThumbs!() * props.step!() * (values().length - 1);
```
Comment (dev-rb) [inline packages/core/src/slider/create-slider-state.ts:127]:
```suggestion
          props.minStepsBetweenThumbs!() * props.step!()
```

**Comment by royeden** [comment]
@dev-rb I've pushed some of the changes, but I'm currently trying to solve a couple more issues that I've found with the component and I would like your take on how to solve them:

1. After using a pointer event to change the slider (either drag or click), it loses focus. This isn't the expected behavior, as `<input type="range">` gains and keeps focus on pointer interactions.
  My easy fix for this (that I really dislike) is manually restoring focus to the element at the end of the `onSlideEnd` call, but we also need to know which thumb to focus and that's kind of tricky. My current approach gets the `fieldId` from the `formContext` and then manually triggers the focus (therefore it always gives focus to the first thumb).
  Let me know how to proceed with this, as it also ties into
2. I'm not sure if having a focusable `span` element as a slider thumb is the best approach, perhaps what's needed is to forward the focus to the input instead? This would simplify event definitions, because anything related to focus / keys is sent to the input, focus styles could still be applied by targeting the internal `input` and label clicks would focus the input correctly.

**Comment by dev-rb** [comment]
@royeden The changes look good, thanks!

As for the issues you've brought up, I'm not too sure what the issue is with focusing. The thumb is given focus when you use it and if you want to focus it when clicking on the `SliderTrack` then the change can be to use the `thumbs` refs signal and the passed index in `onSlideStart` to focus it.

Is the concern here that the `input` should be given focus over the thumb? Or do you mean that the `SliderRoot` should have focus?

**Comment by royeden** [comment]
Oh, I didn't notice the `thumbs` signal, maybe it could be renamed to `thumbRefs` to provide more context for its usage.

I have to check if that allows me to solve the specific issue of the thumb focus, so I'll fix that as soon as I can (probably during these following days 😄 ).

@dev-rb  The thing that I've mentioned in relation to the existence of both inputs and thumbs is more related to a specific issue: the focus is set to the native `input` element if the `Slider.Label` element is clicked, because that's the target for the `label`'s `for` attribute. And that got me thinking if considering using the native elements as the base of interactions instead of the thumbs makes sense, as they respond to some of the basic accessibility concerns natively by default (with the caveat that sometimes native implementations may need a bump of JS to function in a standardized fashion or cover some special use cases).

I've also noticed that the same strategy used for validating the `Home/End` keys via `getThumbMinValue/getThumbMaxValue` could also be applied to each `input` natively via the `min/max` attributes (They are currently set to the global `min/max` of the `Slider`).

**Comment by royeden** [comment]
And what I meant is that the thumb that moved has to be the one that has focus after the move is finished, as it currently doesn't focus anything at all

**Comment by dev-rb** [comment]
Maybe I'm misunderstanding something. The `SliderInput` uses the `thumb` for it's `min/max` values not the global min/max.

As far as converting to use the native `input` element, that I am not too sure about because I largely followed React-ARIA and Radix implementations (as does most of Kobalte). Both of those have heavily put in research and work to handle accessibility. So I'm not too sure if straying from that will be a benefit. Also, from what I understand, implementations for native elements vary across browsers which may raise concern for accessibility and maintenance. 

For the thumb focus, I may also be misunderstanding, but from what I have seen the thumb that you move still contains focus. I only did a simple test of attaching an `onBlur` callback to check if the thumb loses focus. Could be a browser thing maybe?

Please let me know if I misunderstood something @royeden.

**Comment by royeden** [comment]
@dev-rb fixed the thumb focus issue!

I've also noticed that `getClosestValueIndex` wasn't resolving indexes when both thumbs were set to the same value, so I've attempted to solve that by figuring the direction of the `nextValue` in relation to the value of the closest index.

**Comment by royeden** [comment]
I've also added the code to remove [legacy KeyboardEvent.key navigation keys](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#navigation_keys) (this is on the last PR and it can be reverted easily, but I wanted to push the change so it could be reviewed).

**Comment by dev-rb** [comment]
Looks good @royeden! 

For the navigation keys, I don't think removing the legacy keys adds any benefit. Better to be safe in case an issue comes up in the future 😅

**Comment by royeden** [comment]
Got it, will revert that change!

**Comment by royeden** [comment]
@dev-rb That should be all!

**Comment by dev-rb** [comment]
Awesome work @royeden! Thanks for the PR 😄

@fabien-ml This looks ready to be merged

**Comment by royeden** [comment]
Hello @dev-rb just saw the other PR trying to address this issue!

Has this not been deployed yet?

**Comment by dev-rb** [comment]
@royeden There was an error in the last release and it didn't get put in. It should be deployed in the next release!

**Pull request merged**